third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -256,33 +258,68 @@ void dds_device::impl::on_option_value( nlohmann::json const & j, eprosima::fast
     {
         throw std::runtime_error( "control is not an object" );
     }
-    std::string option_name;
-    if( ! rsutils::json::get_ex( control, option_name_key, &option_name ) )
-    {
-        throw std::runtime_error( "missing control/option-name" );
-    }
-    
-    // Find the option and set its value
+
+    // Find the relevant (stream) options to update
     dds_options const * options = &_options;
     std::string stream_name;  // default = empty = device option
-    if( rsutils::json::get_ex( control, stream_name_key, &stream_name ) && !stream_name.empty() )
+    if( rsutils::json::get_ex( control, stream_name_key, &stream_name ) && ! stream_name.empty() )
     {
         auto stream_it = _streams.find( stream_name );
         if( stream_it == _streams.end() )
+            throw std::runtime_error( "stream not found" );
+        options = &stream_it->second->options();
+    }
+
+    auto update_option = [&]( std::string const & option_name, float const new_value )
+    {
+        // Find the option and set its value
+        for( auto & option : *options )
         {
-            throw std::runtime_error( "owner not found" );
+            if( option->get_name() == option_name )
+            {
+                option->set_value( new_value );
+                return;
+            }
         }
-        options = &stream_it->second->options();
+        LOG_DEBUG( "[" << debug_name() << "] option '" << option_name << "': not found" );
+    };
+
+    rsutils::json::nested value_j( j, value_key );
```

Comment:
Can we break this long (and rather complicated) function?
Make it call - update_option, update_options_array etc...

Reply:
The whole function is 80 lines and all three use-cases reuse the lambdas that are common. It's really not **that** complicated.
I can put comments that explain which use-case we're dealing with but I don't want to complicate it even more with different functions, arguments, etc.:

```
        // Without a 'value', the only use-case we have is for a query-option with "option-name":[] (empty array)
        // that returns option-value pairs in 'option-values':
```

```
        // The 'option-name' should be an array of options for which values are returned in a similarly-sized array
```

```
    // A single value for a single option: option-name should be a string
```

Reply:
The comments will help, please add them